model-scoring-core/src/main/scala/org/trustedanalytics/scoring/ScoringService.scala

```
@@ -129,37 +138,103 @@ class ScoringService(model: Model) extends Directives {
+
+              ret.getOrElse(complete(StatusCodes.InternalServerError, "Unable to create the model"))
+            }
+          }
+        }
       }
   }
 
   def scoreStringRequest(records: Seq[Array[Any]]): Array[Any] = {
     records.map(row => {
-      val score = model.score(row)
+      require(scoringModel != null, "Model is null. Please use the upload API to add the model to the Scoring Engine")
+      val score = scoringModel.score(row)
       score(score.length - 1).toString
     }).toArray
   }
 
   def scoreJsonRequest(records: Seq[Array[Any]]): Array[Map[String, Any]] = {
-    records.map(row => scoreToMap(model.score(row))).toArray
+    records.map(row => {
+      require(scoringModel != null, "Model is null. Please use the upload API to add the model to the Scoring Engine")
+      scoreToMap(scoringModel.score(row))
+    }).toArray
   }
 
   def scoreToMap(score: Array[Any]): Map[String, Any] = {
-    val outputNames = model.output().map(o => o.name)
+    val outputNames = scoringModel.output().map(o => o.name)
     require(score.length == outputNames.length, "Length of output values should match the output names")
     val outputMap: Map[String, Any] = outputNames.zip(score).map(combined => (combined._1.name, combined._2)).toMap
     outputMap
   }
+
+  /**
+   * Store the MAR file locally and load the model saved at the given path
+   * @return Model running inside the scoring engine instance
+   */
+  private def installModel(modelBytes: Array[Byte]): String = {
+    if (scoringModel == null) {
+      var tempMarFile: File = null
+      try {
+        tempMarFile = File.createTempFile("model", ".mar")
+        FileUtils.writeByteArrayToFile(tempMarFile, modelBytes)
+        scoringModel = ModelArchiveFormat.read(tempMarFile, this.getClass.getClassLoader, None)
+        "OK"
+      }
+      finally {
+        sys.addShutdownHook(FileUtils.deleteQuietly(tempMarFile)) // Delete temporary File on exit
+      }
+    }
+    else{
+      "Fail"
```

Consider raising an exception instead of returning a string that says "Fail"